### PR TITLE
James 1917 Make documentation in Guice project easier to follow

### PR DESCRIPTION
**Pull request by chibenwa**
I had to help Tuan Le Cong yesterday about starting a James out of docker.

It turned out : 
 - We can not assume JS people can use maven out of the box
 - A configuration we do not run will be outdated. I finished asking him to use the one in docker
 - Working directory needs to contain a folder named conf

As a conlusion, I decided not to have sample configuration when we already have docker containers. We can point to docker configuration. I also made the documentation a bit more clear. I guess...

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Sharing .m2 is not always cool....

```
[Building Java 8 james-1917-8e62a0e8-8409-475f-851f-7f028448d424] [INFO] -------------------------------------------------------------
[Building Java 8 james-1917-8e62a0e8-8409-475f-851f-7f028448d424] [ERROR] COMPILATION ERROR : 
[Building Java 8 james-1917-8e62a0e8-8409-475f-851f-7f028448d424] [INFO] -------------------------------------------------------------
[Building Java 8 james-1917-8e62a0e8-8409-475f-851f-7f028448d424] [ERROR] /james-project/server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/cucumber/CassandraStepdefs.java:[28,43] package org.apache.james.backends.cassandra does not exist
[Building Java 8 james-1917-8e62a0e8-8409-475f-851f-7f028448d424] [ERROR] /james-project/server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/cucumber/CassandraStepdefs.java:[47,13] cannot find symbol
[Building Java 8 james-1917-8e62a0e8-8409-475f-851f-7f028448d424]   symbol:   class EmbeddedCassandra
[Building Java 8 james-1917-8e62a0e8-8409-475f-851f-7f028448d424]   location: class org.apache.james.jmap.cassandra.cucumber.CassandraStepdefs
[Building Java 8 james-1917-8e62a0e8-8409-475f-851f-7f028448d424] [INFO] 2 errors 
```

File: server/container/guice/cassandra-guice/README.adoc [hunk 1]
```diff
@@ -26,20 +32,25 @@ An usual compilation using maven will produce two artifacts into target director
...
 
 You also need to generate a keystore with the following command :
 [source]
 ----
-$ keytool -genkey -alias james -keyalg RSA -keystore conf/keystore
+$ keytool -genkey -alias james -keyalg RSA -keystore sample-configuration/keystore
 ----
 
-You need to have a Cassandra and an ElasticSearch instance running.
+You need to have a Cassandra and an ElasticSearch instance running. You can either launch them via docker or install the servers :
```
Comment (rouazana) [inline server/container/guice/cassandra-guice/README.adoc]: you should invert the or clause, it let think that the next commands are installing the servers

File: server/container/guice/cassandra-guice/README.adoc [hunk 2]
```diff
@@ -26,20 +32,25 @@ An usual compilation using maven will produce two artifacts into target director
...
-A sample directory (appropriately named sample-directory) is provided with some
-default value you may need to replace.
+James requires the configuration to be in a subfolder of working directory that is called **conf**. You can get a sample
+directory for configuration from dockerfiles/run/guice/cassandra/destination/conf. You might need to adapt it to your needs.
 
 You also need to generate a keystore with the following command :
 [source]
 ----
-$ keytool -genkey -alias james -keyalg RSA -keystore conf/keystore
+$ keytool -genkey -alias james -keyalg RSA -keystore sample-configuration/keystore
```
Comment (rouazana) [inline server/container/guice/cassandra-guice/README.adoc]: is it an existing directory?

File: server/container/guice/jpa-guice/README.adoc [hunk 3]
```diff
@@ -12,19 +12,23 @@
...
- * james-server-jpa-guice-${version}.lib
+ * james-server-jpa-guice.jar
+ * james-server-jpa-guice.lib
 
-== Running
+You can for example run in the base of this git repository :
 
-=== Requirements
+```
+mvn clean install -Dskiptests
```
Comment (rouazana) [inline server/container/guice/jpa-guice/README.adoc]: no maven  profile option?

File: server/container/guice/memory-guice/README.adoc [hunk 4]
```diff
@@ -15,14 +15,21 @@ An usual compilation using maven will produce two artifacts into target director
  * james-server-memory-guice-${version}.jar
  * james-server-memory-guice-${version}.lib
 
+ You can for example run in the base of this git repository :
+
+ ```
+ mvn clean install -Dskiptests
```
Comment (rouazana) [inline server/container/guice/memory-guice/README.adoc]: same question

File: server/container/guice/cassandra-guice/README.adoc [hunk 5]
```diff
@@ -5,15 +5,21 @@
...
 
- * james-server-cassandra-guice-${version}.jar
- * james-server-cassandra-guice-${version}.lib
+ * james-server-cassandra-guice.jar
+ * james-server-cassandra-guice.lib
+
+You can for example run in the base of this git repository :
+
+```
+mvn clean install -Dskiptests
```
Comment (mbaechler) [inline server/container/guice/cassandra-guice/README.adoc]: please don't suggest the default is skipping  test (:

Comment (chibenwa) [inline server/container/guice/cassandra-guice/README.adoc]: Nope, profiles only enable MPT tests / JMAP integration tests / webadmin tests. To produce the jars, this is enougth

File: server/container/guice/cassandra-guice/README.adoc (hunk 2 above)
Comment (chibenwa) [inline server/container/guice/cassandra-guice/README.adoc]: Yes

File: server/container/guice/jpa-guice/README.adoc (hunk 3 above)
Comment (chibenwa) [inline server/container/guice/jpa-guice/README.adoc]: Nope, as stated before

File: server/container/guice/memory-guice/README.adoc (hunk 4 above)
Comment (chibenwa) [inline server/container/guice/memory-guice/README.adoc]: Same answer

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**